From fdfd91b8372bcae5e8a44553a11804c393b11e09 Mon Sep 17 00:00:00 2001
From: Ali B <abittin@gmail.com>
Date: Thu, 13 Dec 2018 17:05:01 +0300
Subject: [PATCH 10/18] base: fix mediascanner access permissions to external
 storage

12-13 16:52:37.579  2621  5891 E DatabaseUtils: Writing exception to parcel

12-13 16:52:37.579  2621  5891 E DatabaseUtils: java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/audio/media/10579 from pid=1270, uid=1000 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()

12-13 16:52:37.579  2621  5891 E DatabaseUtils:         at android.content.ContentProvider.enforceReadPermissionInner(ContentProvider.java:634)

12-13 16:52:37.579  2621  5891 E DatabaseUtils:         at com.android.providers.media.MediaProvider.enforceReadPermissionInner(MediaProvider.java:705)

12-13 16:52:37.579  2621  5891 E DatabaseUtils:         at android.content.ContentProvider$Transport.enforceReadPermission(ContentProvider.java:503)

12-13 16:52:37.579  2621  5891 E DatabaseUtils:         at android.content.ContentProvider$Transport.enforceFilePermission(ContentProvider.java:494)

12-13 16:52:37.579  2621  5891 E DatabaseUtils:         at android.content.ContentProvider$Transport.openFile(ContentProvider.java:371)

12-13 16:52:37.579  2621  5891 E DatabaseUtils:         at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:229)

12-13 16:52:37.579  2621  5891 E DatabaseUtils:         at android.os.Binder.execTransact(Binder.java:731)

Change-Id: I093d544a7cfa755bf9aa1704ebb1cfd2f919325c
---
 .../server/pm/permission/DefaultPermissionGrantPolicy.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java b/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java
index a4e4f03309e7..efc3c3fc55ec 100644
--- a/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java
+++ b/services/core/java/com/android/server/pm/permission/DefaultPermissionGrantPolicy.java
@@ -983,6 +983,13 @@ public final class DefaultPermissionGrantPolicy {
         if (mPermissionGrantedCallback != null) {
             mPermissionGrantedCallback.onDefaultRuntimePermissionsGranted(userId);
         }
+
+        // Mediascanner
+        PackageParser.Package mediascannerPackage = getDefaultProviderAuthorityPackage(
+                "com.android.providers.media.MediaProvider", userId);
+        if (mediascannerPackage != null) {
+            grantRuntimePermissions(mediascannerPackage, STORAGE_PERMISSIONS, true, userId);
+        }
     }
 
     private void grantDefaultPermissionsToDefaultSystemDialerApp(
-- 
2.17.1

